primitives/avn-common/src/event_discovery.rs

```diff
@@ -54,33 +68,43 @@ impl Ord for DiscoveredEvent {
     }
 }
 
+pub type FractionsCount = u16;
 type EthEventsPartition = BoundedBTreeSet<DiscoveredEvent, EventsBatchLimit>;
 #[derive(PartialEq, Eq, Clone, Encode, Decode, Debug, TypeInfo, MaxEncodedLen)]
 pub struct DiscoveredEthEventsFraction {
-    data: EthEventsPartition,
-    fraction: u16,
-    fraction_count: u16,
     id: H256,
+    fraction: FractionsCount,
+    fraction_count: FractionsCount,
+    data: EthEventsPartition,
 }
 
 impl DiscoveredEthEventsFraction {
-    pub fn events(&self) -> &EthEventsPartition {
-        &self.data
+    pub fn id(&self) -> &H256 {
+        &self.id
     }
 
-    pub fn fraction(&self) -> u16 {
+    pub fn fraction(&self) -> FractionsCount {
```

Comment:
whats the difference between `fraction` and `fraction_count`?

Reply:
fraction_count is how many fractions has this broken into. Fraction is which of the fraction is it.
Like saying 1 out of 5 fractions